RXPromise Libraries/Source/RXPromise.mm

```diff
@@ -380,12 +374,11 @@ - (void)synced_updateWithProgress:(float)progress {
             }
         }
     }
-    if (_children) {
-        for(const RXPromise * __weak child : *_children) {
+    if (_children.size()) {
```

Comment:
`if (_children.size())` may be omitted. Just iterating over an empty list should work, too.